### PR TITLE
[HttpKernel] Fix timezone influenced tests

### PR DESCRIPTION
**Pull request by GromNaN**
| Q             | A
| ------------- | ---
| Branch?       | 6.1
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | https://github.com/symfony/symfony/pull/45589#issuecomment-1079789237
| License       | MIT
| Doc PR        | -

`DateTimeValueResolverTest` fails depending on the default timezone and the current time.

The culprit is the combinaison of `strtotime` + `new DateTime`.
https://github.com/symfony/symfony/blob/24304ad77874f03f9971fd9e29c0ef0776b0f5b9/src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/DateTimeValueResolver.php#L67-L69

```
There were 3 failures:

1) Symfony\Component\HttpKernel\Tests\Controller\ArgumentResolver\DateTimeValueResolverTest::testFullDate
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'2012-07-21'
+'2012-07-20'

/home/runner/work/symfony/symfony/src/Symfony/Component/HttpKernel/Tests/Controller/ArgumentResolver/DateTimeValueResolverTest.php:53

2) Symfony\Component\HttpKernel\Tests\Controller\ArgumentResolver\DateTimeValueResolverTest::testCustomClass
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'2016-09-08'
+'2016-09-07'

/home/runner/work/symfony/symfony/src/Symfony/Component/HttpKernel/Tests/Controller/ArgumentResolver/DateTimeValueResolverTest.php:100

3) Symfony\Component\HttpKernel\Tests\Controller\ArgumentResolver\DateTimeValueResolverTest::testDateTimeImmutable
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'2016-09-08'
+'2016-09-07'

/home/runner/work/symfony/symfony/src/Symfony/Component/HttpKernel/Tests/Controller/ArgumentResolver/DateTimeValueResolverTest.php:116
```

### CODE REVIEW

File: src/Symfony/Component/HttpKernel/Tests/Controller/ArgumentResolver/DateTimeValueResolverTest.php [hunk 1]
```diff
@@ -149,6 +171,16 @@ public function test404Exception(ArgumentMetadata $argument, Request $request)
         iterator_to_array($results);
     }
 
+    protected function setUp(): void
```
Comment (fabpot) [inline src/Symfony/Component/HttpKernel/Tests/Controller/ArgumentResolver/DateTimeValueResolverTest.php]: Should be the first methods in the class to be consistent with our CS.

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/HttpKernel/Tests/Controller/ArgumentResolver/DateTimeValueResolverTest.php (hunk 1 above)
Comment (GromNaN) [reply src/Symfony/Component/HttpKernel/Tests/Controller/ArgumentResolver/DateTimeValueResolverTest.php]: Fixed.

**Review by yceruto — APPROVED** [review]

**Review by codedmonkey — APPROVED** [review]

**Comment by codedmonkey** [comment]
Sorry about that, I wasn't clear in my PR and my initial fix was insufficient so I rolled it back.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @GromNaN.

**Pull request merged**